inc/vcf/fixer.hpp

```diff
@@ -156,11 +157,68 @@ namespace ebi
             util::writeline(output, *line);
             ignored_errors++;
         }
+
+        /**
+         * explanation of the fix:
+         * - in error.get_field there will be the field name as it appears in the FORMAT column, e.g. "GT".
+         * - don't do anything if error.get_field() is empty
+         * - get the index in the FORMAT column of error.get_field(), e.g. in GT:AC:AN, the index of AC is 1.
+         * - for each sample column
+         *      - put a "." instead of the value in the index-th position of ":" e.g. 0/0:3:4 becomes 0/0:.:4 for index = 1
+         * @param error
+         */
         virtual void visit(SamplesBodyError &error) override
         {
-            util::writeline(output, *line);
-            ignored_errors++;
+            if (error.get_field().empty()) {
+                util::writeline(output, *line);
+                ignored_errors++;
+            } else {
+                std::cerr << "DEBUG: line " << error.get_line() << ": fixing invalid sample field " << error.get_field()
+                          << std::endl;
+
+                const size_t format_column = 8;
+//                size_t first_samples_column = 9;
+                const char empty_subfield = '.';
+                std::string string_line = {line->begin(), line->end()};
+
+                using iter = std::vector<std::string>::iterator;
+                size_t fixed_samples = fix_columns(format_column, -1, string_line, "\t", [&](iter first, iter last) {
+
+                    size_t subfield_index = split_and_find(*first, ":", error.get_field());
+                    if (subfield_index == first->npos) {
+                        std::cerr << "WARNING: line " << error.get_line() << ": tried to fix field "
+                                  << error.get_field() << " but it was not present in the FORMAT column";
+                        ignored_errors++;
+                        return;
+                    }
+                    output << *first;   // write the FORMAT column
+
+                    // if the field is GT, the values must use "/", as in "./."
+                    const std::string subfield_separator = subfield_index == 0? "/" : ",";
+
+                    // error.number should be -1 if the cardinality is unknown, and any positive number otherwise
+                    // so, if unknown, put 1, so that a single "." is written
+                    size_t number = error.get_number() <= -1? 1 : static_cast<size_t>(error.get_number());
```

Comment:
Could these `number` variables be renamed to something like `field_cardinality`? It is not clear at first sight what their purpose is.